File: dart/hamming/lib/hamming.dart
class Hamming {
  int compute(String lhs, String rhs) {
    if (lhs.length != rhs.length) throw new ArgumentError();
    int ham = 0;
    for (int index = 0; index < lhs.length; index++)
      if (lhs[index] != rhs[index]) ham++;
    return ham;
  }
}

File: kotlin/hamming/src/main/kotlin/Hamming.kt
object Hamming {

    fun compute(leftStrand: String, rightStrand: String): Int {
        require(leftStrand.length == rightStrand.length)
            {"left and right strands must be of equal length"}
        
        val maxIndex = leftStrand.length - 1
        var hammingNumber = 0

        for (index in 0..maxIndex)
            if (leftStrand[index] != rightStrand[index])
                hammingNumber++

        return hammingNumber
    }
}
